src/mlpack/core/data/split_data.hpp

```diff
@@ -18,11 +18,108 @@
 namespace mlpack {
 namespace data {
 
+template<typename InputType, typename LabelsType>
+void SplitHelper(const InputType& input,
+           const LabelsType& inputLabel,
+           InputType& trainData,
+           InputType& testData,
+           LabelsType& trainLabel,
+           LabelsType& testLabel,
+           const double testRatio,
+           const bool shuffleData = true)
+{
+  const size_t testSize = static_cast<size_t>(input.n_cols * testRatio);
+  const size_t trainSize = input.n_cols - testSize;
+
+  trainData.set_size(input.n_rows, trainSize);
+  testData.set_size(input.n_rows, testSize);
+  trainLabel.set_size(inputLabel.n_rows, trainSize);
+  testLabel.set_size(inputLabel.n_rows, testSize);
+
+  if (shuffleData)
+  {
+    arma::uvec order = arma::shuffle(arma::linspace<arma::uvec>(0,
+        input.n_cols - 1, input.n_cols));
+
+    if (trainSize > 0)
+    {
+      for (size_t i = 0; i < trainSize; ++i)
+      {
+        trainData.col(i) = input.col(order(i));
+        trainLabel.col(i) = inputLabel.col(order(i));
+      }
+    }
+    if (trainSize < input.n_cols)
+    {
+      for (size_t i = trainSize; i < input.n_cols; ++i)
+      {
+        testData.col(i - trainSize) = input.col(order(i));
+        testLabel.col(i - trainSize) = inputLabel.col(order(i));
+      }
+    }
+  }
+  else
+  {
+    if (trainSize > 0)
+    {
+      trainData = input.cols(0, trainSize - 1);
+      trainLabel = inputLabel.cols(0, trainSize - 1);
+    }
+
+    if (trainSize < input.n_cols)
+    {
+      testData = input.cols(trainSize, input.n_cols - 1);
+      testLabel = inputLabel.cols(trainSize, inputLabel.n_cols - 1);
+    }
+  }
+}
+
+template<typename InputType>
+void SplitHelper(const InputType& input,
```

Comment:
What about we add the `order` vector as an optional parameter as well, that way, you could  reuse the function for the case above. Let me know what you think.

Reply:
Sorry, I didn't understand you exactly. `order` is being used in both the functions. Did you mean `labels` ? Correct me if I overlooked.

Reply:
Another idea would be to have one helper function that splits whatever you give it, then call it twice, once with the data and once with the labels.

Reply:
Thats a great idea. I will try to do that 👍